### PR TITLE
tailscale: update tailfs functions and vars to use drive naming

### PR DESCRIPTION
**Pull request by catzkorn**
This change updates all tailfs functions and the majority of the tailfs variables to use the new drive naming.

Update tailscale/corp#16827

### CODE REVIEW

File: client/tailscale/localclient.go [hunk 1]
```diff
@@ -1418,25 +1418,25 @@ func (lc *LocalClient) CheckUpdate(ctx context.Context) (*tailcfg.ClientVersion,
 	return &cv, nil
 }
 
-// TailFSSetFileServerAddr instructs TailFS to use the server at addr to access
+// DriveSetFileServerAddr instructs drive to use the server at addr to access
```
Comment (bradfitz) [inline client/tailscale/localclient.go:1421]: drop File in FileServer?

Drive not capitalized? If it's not a proper noun then "instructs drive to use" is weird lowercase.

File: client/tailscale/localclient.go [hunk 2]
```diff
@@ -1418,25 +1418,25 @@ func (lc *LocalClient) CheckUpdate(ctx context.Context) (*tailcfg.ClientVersion,
...
 // the filesystem. This is used on platforms like Windows and MacOS to let
-// TailFS know to use the file server running in the GUI app.
-func (lc *LocalClient) TailFSSetFileServerAddr(ctx context.Context, addr string) error {
+// drive know to use the file server running in the GUI app.
+func (lc *LocalClient) DriveSetFileServerAddr(ctx context.Context, addr string) error {
```
Comment (bradfitz) [inline client/tailscale/localclient.go:1424]: What's the addr like? An IP:port? A URL?

File: wgengine/netstack/netstack.go [hunk 3]
```diff
@@ -670,7 +670,7 @@ func (ns *Impl) handleLocalPackets(p *packet.Parsed, t *tstun.Wrapper) filter.Re
 		return filter.DropSilently
 	}
 
-	// If it's not traffic to the service IP (e.g. magicDNS or TailFS) we don't
+	// If it's not traffic to the service IP (e.g. magicDNS or Taildrive) we don't
```
Comment (bradfitz) [inline wgengine/netstack/netstack.go:673]: Taildrive capitalized is good. Do this for the lowercase "drive" elsewhere?

File: client/tailscale/localclient.go (hunk 2 above)
Comment (oxtoacart) [reply client/tailscale/localclient.go:1424]: It's `ip:port`.

File: client/tailscale/localclient.go (hunk 1 above)
Comment (oxtoacart) [reply client/tailscale/localclient.go:1421]: My preference would be to use `drive/Drive` for variable and package names, but in comments to refer to it as `Taildrive`. That's what I did [here](https://github.com/tailscale/corp/pull/18871).

File: wgengine/netstack/netstack.go (hunk 3 above)
Comment (oxtoacart) [reply wgengine/netstack/netstack.go:673]: If we always refer to it as "Taildrive" in comments, capitalization of "drive" becomes moot.

**Comment by oxtoacart** [comment]
Is it worth making this PR the one to remove all TailFS references? `grep -ri tailfs .` still brings up a lot of hits.

**Comment by catzkorn** [comment]
> Is it worth making this PR the one to remove all TailFS references? `grep -ri tailfs .` still brings up a lot of hits.

The remaining references are error messages, Paths, nodecaps, etc. I'm going to update all of those in the next PR, since there is logical/behavioral changes involved.

**Review by creachadair — APPROVED** [review]

**Pull request merged**